pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -852,6 +856,15 @@ func (gce *GCECloud) createTargetPool(name, serviceName, region string, hosts []
 	for _, host := range hosts {
 		instances = append(instances, makeHostURL(gce.projectID, host.Zone, host.Name))
 	}
+	if len(instances) > maxInstancesPerTargetPool {
```

Comment:
>It's probably the right thing to do, but I feel like this is moot eventually and we don't filter for healthy nodes today...

Actually we already only pass in ready nodes (https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/service/servicecontroller.go#L655). So what I was recommending is unnecessary.